packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+      };
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.setup.mockRejectedValueOnce(new Error());
+
+      await expect(
+        escrowClient.setup(ethers.constants.AddressZero, escrowConfig)
+      ).rejects.toThrow();
+
+      expect(escrowClient.escrowContract.setup).toHaveBeenCalledWith(
+        ethers.constants.AddressZero,
+        ethers.constants.AddressZero,
+        BigNumber.from(50),
+        BigNumber.from(50),
+        VALID_URL,
+        FAKE_HASH
+      );
+    });
+  });
+
+  describe('createAndSetupEscrow', () => {
+    test('should successfully create and setup escrow', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const tokenAddress = ethers.constants.AddressZero;
+      const trustedHandlers = [ethers.constants.AddressZero];
+      const escrowConfig = {
+        recordingOracle: ethers.constants.AddressZero,
+        reputationOracle: ethers.constants.AddressZero,
+        recordingOracleFee: BigNumber.from(50),
+        reputationOracleFee: BigNumber.from(50),
+        manifestUrl: VALID_URL,
+        hash: FAKE_HASH,
+      };
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.createEscrow = vi.fn().mockReturnValue(escrowAddress);
+      escrowClient.escrowContract.setup.mockReturnValue(true);
+
+      await escrowClient.createAndSetupEscrow(
+        tokenAddress,
+        trustedHandlers,
+        escrowConfig
+      );
+
+      expect(escrowClient.createEscrow).toHaveBeenCalledWith(
+        tokenAddress,
+        trustedHandlers
+      );
+      expect(escrowClient.escrowContract.setup).toHaveBeenCalledWith(
+        ethers.constants.AddressZero,
+        ethers.constants.AddressZero,
+        BigNumber.from(50),
+        BigNumber.from(50),
+        VALID_URL,
+        FAKE_HASH
+      );
+    });
+
+    test('should throw an error if the setup escrow fails', async () => {
```

fix writing: 
'should throw an error if setup escrow fails'